pyro/infer/reparam/stable.py

```diff
@@ -94,7 +100,7 @@ def __call__(self, name, fn, obs):
         half_pi = proto.new_full(proto.shape, math.pi / 2)
         one = proto.new_ones(proto.shape)
         u = pyro.sample("{}_uniform".format(name),
-                        self._wrap(dist.Uniform(-half_pi, half_pi), event_dim))
+                        self._wrap(dist.Uniform(-half_pi + self.epsilon, half_pi - self.epsilon), event_dim))
```

Comment:
it's likely similar effects could be achieved by instead modifying computations in `_unsafe_standard_stable`.

perhaps `(v - V).cos()`?  (i haven't thought through the math in much detail)

Reply:
This looks like the right direction. I’ll take a look.

Reply:
You pointed out the right place. It would be better to use
```
(v - V).cos().clamp_(min=torch.finfo(v.dtype).tiny)
```
This is precision issue
```
alpha = torch.tensor(0.99)
half_pi = math.pi / 2
assert half_pi > (half_pi - (1 - alpha) * 1e-6)
```
fails in single precision; hence cosine of RHS < 0.

Reply:
This seems reasonable. It would be nice to implement it in a way that applies to all stable logic, not only this reparameterizer, e.g. in `_unsafe_standard_stable()` as you suggest. Maybe we can add something like
```py
def _unsafe_standard_stable(..., tol=1e-4):
    if epsilon:
        assert 0 < tol < 1
        u = u * (1 - tol)
    ...
```
and expose `tol` through each reparameterizer and/or the Stable distribution? @fehiepsi WDYT?

Reply:
@fehiepsi great, I'd love to find a solution that did not expose parameters like `tol` and `epsilon`!